Use a selector for platform mode (iOS / Android) instead of a button.

kenzieschmoll opened the pull request:
Addresses https://github.com/flutter/devtools/issues/763 @sfshaza2

Also regrouped the buttons to put this selector at the far right.

![Screen Shot 2019-07-10 at 2 07 54 PM](https://user-images.githubusercontent.com/43759233/61005042-37144d80-a31c-11e9-88a5-b72541f4c05a.png)
![Screen Shot 2019-07-10 at 2 08 08 PM](https://user-images.githubusercontent.com/43759233/61005047-3976a780-a31c-11e9-8e7e-2dd3a72b934a.png)

@jacob314

devoncarew commented:
As a drive by comment, we may want to make it clearer what you're selecting when you select 'iOS' or 'Android'. It's the platform the flutter framework is emulating; perhaps `platform: iOS`? `framework mode: iOS`? We'd want something brief, and familiar to users (based on the flutter framework name for the option?).

jacob314 commented:
Great to see this added. The current behavior is quite confusing given it is really three values.
"No platform override"
"iOS"
"Android"

We could call the dropdown
"Platform override" to approximately match the names used in flutter.
The most technically accurate description is `The platform that user interaction should adapt to target.` but that is of course way too long to be useful.

Lets also consider adding Fuchsia to this list to match the list from `platform.dart` in package:flutter now that we have a list instead of two options.

```dart
/// The platform that user interaction should adapt to target.
///
/// The [defaultTargetPlatform] getter returns the current platform.
enum TargetPlatform {
  /// Android: <https://www.android.com/>
  android,

  /// Fuchsia: <https://fuchsia.googlesource.com/>
  fuchsia,

  /// iOS: <http://www.apple.com/ios/>
  iOS,
}
```

kenzieschmoll commented:
Added fuchsia and changed display to "Platform: Android". Also added support for multi-state extensions. Up to this point we have only handled 2-state extensions (enabled / disabled).

jacob314 commented:
I just finally landed the version pin so you should be able to remove it from this CL.

jacob314 commented on packages/devtools/lib/src/service_extensions.dart:
```diff
@@ -10,31 +10,72 @@ import 'ui/analytics_constants.dart' as ga;
     @required this.gaScreenName,
     @required this.gaItem,
-  });
+  }) : displayValues =
```
add
assert(values.length == tooltips.length)

kenzieschmoll replied:
This isn't necessarily the case. For example, the platform selector just has one tooltip for the selector ("Override Target Platform"), not one for each of the items in the selection list. For the 2-state extensions, `values.length == tooltips.length` is enforced by the constructor of `ToggleableServiceExtensionDescription`.

jacob314 commented on packages/devtools/lib/src/service_extensions.dart:
```diff
@@ -118,19 +159,18 @@ const slowAnimations = ToggleableServiceExtensionDescription<num>._(
-  disabledValue: 'android',
-  enabledTooltip: 'Toggle iOS Platform',
-  disabledTooltip: 'Toggle iOS Platform',
+  values: ['iOS', 'android', 'fuchsia'],
```
we should also have a value that is 
"unset" or similar
That is the default state where the value actually matches your platform.

kenzieschmoll replied:
The starting value is currently set to whatever the platform actually is.

 I think "Platform: none" or "Platform: unset" could be kind of confusing. Even more confusing would be a dropdown whose starting value is just "Unset". I think having the starting value be consistent with the device is actually more intuitive because it helps connect the dots between the device and what the dropdown selection changes.
